presto-main/src/test/java/com/facebook/presto/operator/TestOperationTimer.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.OperationTimer.OperationTiming;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+
+public class TestOperationTimer
+{
+    public static int blackHole;
+
+    @Test
+    public void testOverallCpuTrackingEnabled()
+    {
+        OperationTiming timing = new OperationTiming();
+
+        OperationTimer timer = new OperationTimer(true, false);
```

Comment:
I think you might be easier to read this if you form it as a loop, which I think works if you start with the previous values set to zero.